glfs.c

```diff
@@ -626,17 +627,35 @@ static int tcmu_glfs_write(struct tcmu_device *dev,
 static int tcmu_glfs_reconfig(struct tcmu_device *dev,
                               struct tcmulib_cfg_info *cfg)
 {
+	struct glfs_state *gfsp = tcmu_get_dev_private(dev);
+	struct stat st;
+	int ret = -1;
+
 	switch (cfg->type) {
 	case TCMULIB_CFG_DEV_SIZE:
-		/*
-		 * Let glusterfs tools handle size checks
-		 */
-		return 0;
+		ret = glfs_lstat(gfsp->fs, gfsp->hosts->path, &st);
+		if (ret) {
+			tcmu_dev_warn(dev, "glfs_lstat failed: %m\n");
+			tcmu_notify_conn_lost(dev);
+			ret = 0;  /* Let the targetcli command return success */
+		} else {
+			if (st.st_size != cfg->data.dev_size) {
+				tcmu_dev_err(dev,
+					     "device size and backing size disagree: "
+					     "device %lld backing %lld\n",
+					     cfg->data.dev_size,
+					     (long long) st.st_size);
+				ret = -1;
+			}
+		}
+		break;
 	case TCMULIB_CFG_DEV_CFGSTR:
 	case TCMULIB_CFG_WRITE_CACHE:
 	default:
 		return -EOPNOTSUPP;
 	}
+
```

Comment:
The ret value should be a -Exyz value like -ENOTCONNN,-EINVAL, etc. The value will get propagated to the caller so it can have a better idea of what happened.

Reply:
Will update.